pkg/registry/types.go

```diff
@@ -176,3 +177,206 @@ type Metadata struct {
 func (m *Metadata) ParsedTime() (time.Time, error) {
 	return time.Parse(time.RFC3339, m.LastUpdated)
 }
+
+// ServerMetadata is an interface that both ImageMetadata and RemoteServerMetadata implement
+type ServerMetadata interface {
+	// GetName returns the server name
+	GetName() string
+	// GetDescription returns the server description
+	GetDescription() string
+	// GetTier returns the server tier
+	GetTier() string
+	// GetStatus returns the server status
+	GetStatus() string
+	// GetTransport returns the server transport
+	GetTransport() string
+	// GetTools returns the list of tools provided by the server
+	GetTools() []string
+	// GetMetadata returns the server metadata
+	GetMetadata() *Metadata
+	// GetRepositoryURL returns the repository URL
+	GetRepositoryURL() string
+	// GetTags returns the server tags
+	GetTags() []string
+	// GetCustomMetadata returns custom metadata
+	GetCustomMetadata() map[string]any
+	// IsRemote returns true if this is a remote server
+	IsRemote() bool
+	// GetEnvVars returns environment variables
+	GetEnvVars() []*EnvVar
+	// GetRawImplementation returns the underlying implementation (either *ImageMetadata or *RemoteServerMetadata)
+	GetRawImplementation() any
+}
+
+// Implement ServerMetadata interface for ImageMetadata
+
+// GetName returns the server name
+func (i *ImageMetadata) GetName() string {
+	return i.Name
+}
+
+// GetDescription returns the server description
+func (i *ImageMetadata) GetDescription() string {
+	return i.Description
+}
+
+// GetTier returns the server tier
+func (i *ImageMetadata) GetTier() string {
+	return i.Tier
+}
+
+// GetStatus returns the server status
+func (i *ImageMetadata) GetStatus() string {
+	return i.Status
+}
+
+// GetTransport returns the server transport
+func (i *ImageMetadata) GetTransport() string {
+	return i.Transport
+}
+
+// GetTools returns the list of tools provided by the server
+func (i *ImageMetadata) GetTools() []string {
+	return i.Tools
+}
+
+// GetMetadata returns the server metadata
+func (i *ImageMetadata) GetMetadata() *Metadata {
+	return i.Metadata
+}
+
+// GetRepositoryURL returns the repository URL
+func (i *ImageMetadata) GetRepositoryURL() string {
+	return i.RepositoryURL
+}
+
+// GetTags returns the server tags
+func (i *ImageMetadata) GetTags() []string {
+	return i.Tags
+}
+
+// GetCustomMetadata returns custom metadata
+func (i *ImageMetadata) GetCustomMetadata() map[string]any {
+	return i.CustomMetadata
+}
+
+// IsRemote returns false for container servers
+func (*ImageMetadata) IsRemote() bool {
+	return false
+}
+
+// GetEnvVars returns environment variables
+func (i *ImageMetadata) GetEnvVars() []*EnvVar {
+	return i.EnvVars
+}
+
+// GetRawImplementation returns the underlying ImageMetadata pointer
+func (i *ImageMetadata) GetRawImplementation() any {
+	return i
```

Comment:
Nit: This doesn't seem very useful. You have to cast from `any` to the type you expect. You could do the same by casting the instance of `ServerMetadata` to the type you expect, and skip this method.